cli/ballerina-go-cli/bal/pkg/utils/utils.go

```diff
@@ -0,0 +1,300 @@
+	return port > 0
+}
+
+func SetBallerinaClassPath(ballerinaHome string, javaHome string) string {
+	var ballerinaClasspath string
+	toolsJarPath := filepath.Join(ballerinaHome, "bre", "lib", "tools.jar")
+	if _, err := os.Stat(toolsJarPath); err == nil {
+		ballerinaClasspath += filepath.Join(javaHome, "lib", "tools.jar")
+	}
+	jarPaths := []string{
+		filepath.Join(ballerinaHome, "bre", "lib"),
+		filepath.Join(ballerinaHome, "lib", "tools", "lang-server", "lib"),
+		filepath.Join(ballerinaHome, "lib", "tools", "debug-adapter", "lib"),
+	}
+	for _, path := range jarPaths {
+		files, _ := filepath.Glob(filepath.Join(path, "*.jar"))
+		for _, f := range files {
+			ballerinaClasspath += string(filepath.ListSeparator) + f
+		}
+	}
+
+	return ballerinaClasspath
+}
+
+func GetJavaOpts() string {
+	var balJavaDebug string
+	balJavaDebug, isSet := os.LookupEnv("BAL_JAVA_DEBUG")
+	javaOpts := ""
+	if isSet {
+		if balJavaDebug == "" {
+			fmt.Println("Please specify the debug port for the BAL_JAVA_DEBUG variable")
+			os.Exit(1)
+		} else {
+			javaOpts = os.Getenv("JAVA_OPTS")
+
+			if javaOpts != "" {
+				fmt.Println("Warning !!! User specified JAVA_OPTS may interfere with BAL_JAVA_DEBUG")
+			}
+
+			balDebugOpts := os.Getenv("BAL_DEBUG_OPTS")
+
+			if balDebugOpts != "" {
+				javaOpts = javaOpts + " " + balDebugOpts
+			} else {
+				javaOpts = javaOpts + " -Xdebug -Xnoagent -Djava.compiler=NONE -Xrunjdwp:transport=dt_socket,server=y,suspend=y,address=" + balJavaDebug
+			}
+		}
+	}
+	return javaOpts
+}
+
+func GetJavaSettings(ballerinaHome string) (javaHome, javaCmd string) {
+	javaPath := filepath.Join(ballerinaHome, "..", "..", "dependencies", "@JDK_VERSION@")
+	javaCmd = os.Getenv("JAVACMD")
+	javaHome = os.Getenv("JAVA_HOME")
+	if stat, err := os.Stat(javaPath); err == nil && stat.IsDir() {
+		javaHome = javaPath
+	}
+	OS := runtime.GOOS
+	if OS == "windows" {
+		javaExePath := filepath.Join(javaHome, "bin", "java.exe")
+		if _, err := os.Stat(javaExePath); err != nil {
+			javaCmd = javaExePath
+		}
+	} else {
+		javaExePath := filepath.Join(javaHome, "bin", "java")
+		if _, err := os.Stat(javaExePath); err != nil {
+			javaCmd = javaExePath
+		}
+	}
+	//Set javaHome to installed java version in the user's device
+	switch OS {
+	case "darwin":
+		javaVersion := os.Getenv("JAVA_VERSION")
+		path := filepath.Join(string(filepath.Separator), "usr", "libexec", "java_home")
+		if javaHome == "" {
+			if javaVersion == "" {
+				CmdOutput, err := exec.Command(path).Output()
+				if err == nil {
+					javaHome = strings.TrimSpace(string(CmdOutput))
+				}
+			} else {
+				// If JAVA_HOME is not set, but JAVA_VERSION is set, get the Java home for the specified version
+				CmdOutput, err := exec.Command(path, "-v", javaVersion).Output()
+				if err == nil {
+					javaHome = strings.TrimSpace(string(CmdOutput))
+				}
+			}
+		}
+
+	}
+
+	if javaCmd == "" {
+		if javaHome != "" {
+			javaCmdPath := filepath.Join(javaHome, "jre", "sh", "java")
```

Comment:
shall we use a constant for `java`? Use constants for other possible places as well.